packages/salesforce-adapter/src/filters/convert_maps.ts

```diff
@@ -416,11 +529,27 @@ const filter: LocalFilterCreator = ({ config }) => ({
         if (instancesToConvert.length === 0) {
           await updateFieldTypes(typeToConvert, [], mapFieldDef)
         } else {
-          const nonUniqueMapFields = await convertInstanceFieldsToMaps(instancesToConvert, mapFieldDef)
+          const nonUniqueMapFields = await convertElementFieldsToMaps(instancesToConvert, mapFieldDef)
           await updateFieldTypes(typeToConvert, nonUniqueMapFields, mapFieldDef)
         }
       }
     })
+
+    const fields = elements.filter(isCustomObjectSync).flatMap(obj => Object.values(obj.fields))
+    await awu(Object.entries(fieldTypeToAnnotationToMapDef)).forEach(async ([fieldType, annotationToMapDef]) => {
+      if (
+        ['Picklist', 'MultiselectPicklist'].includes(fieldType) &&
+        !config.fetchProfile.isFeatureEnabled('picklistsAsMaps')
+      ) {
+        return
+      }
+      const fieldsToConvert = fields.filter(field => field.refType.elemID.typeName === fieldType)
+      if (fieldsToConvert.length === 0) {
+        return
+      }
+      const nonUniqueMapFields = await convertElementFieldsToMaps(fieldsToConvert, annotationToMapDef)
+      await updateFieldTypes(await fieldsToConvert[0].getType(), nonUniqueMapFields, annotationToMapDef)
+    })
```

Comment:
Isn't this logic only relevant for `Picklists` right now? 🤔 
```suggestion
if (config.fetchProfile.isFeatureEnabled('picklistsAsMaps')) {
    const fields = elements.filter(isCustomObjectSync).flatMap(obj => Object.values(obj.fields))
    await awu(Object.entries(fieldTypeToAnnotationToMapDef)).forEach(async ([fieldType, annotationToMapDef]) => {
      const fieldsToConvert = fields.filter(field => field.refType.elemID.typeName === fieldType)
      if (fieldsToConvert.length === 0) {
        return
      }
      const nonUniqueMapFields = await convertElementFieldsToMaps(fieldsToConvert, annotationToMapDef)
      await updateFieldTypes(await fieldsToConvert[0].getType(), nonUniqueMapFields, annotationToMapDef)
    })
 }
```

Reply:
Yes, but why not structure the code to allow more usage alonside picklists without refactoring the code? IMO the current version is more accurate - only picklist changes should be gated by the `picklistsAsMaps` FF.